rotkehlchen/api/v1/resources.py

```diff
@@ -2370,26 +2372,27 @@ def post(self, file: FileStorage) -> Response:
         return response
 
 
-class DBSnapshotExportingResource(BaseMethodView):
-    post_schema = SnapshotExportingSchema()
+class DBSnapshotsResource(BaseMethodView):
+    get_schema = SnapshotQuerySchema()
+    upload_schema = SnapshotImportingSchema()
+    patch_schema = SnapshotEditingSchema()
+    delete_schema = SnapshotTimestampQuerySchema()
 
     @require_loggedin_user()
-    @use_kwargs(post_schema, location='json')
-    def post(self, timestamp: Timestamp, path: Path) -> Response:
-        return self.rest_api.export_user_db_snapshot(timestamp=timestamp, path=path)
-
+    @use_kwargs(get_schema, location='json_and_query_and_view_args')
+    def get(
+            self,
+            action: Optional[str],
```

Comment:
Shouldn't thetype of this be `Optional[Literal['x', 'y', ..]]` where 'x' and 'y' are the action values?